stages/org.osbuild.oscap.remediation

```diff
@@ -89,34 +90,53 @@ SCHEMA = """
 }
 """
 
-# Map containing the translations between
-# our configuration names and the OpenScap
-# configuration names.
-OSCAP_OPTIONS = {
-    "datastream" : "OSCAP_REMEDIATE_DS",
-    "profile_id" : "OSCAP_REMEDIATE_PROFILE_ID",
-    "datastream_id" : "OSCAP_REMEDIATE_DATASTREAM_ID",
-    "xccdf_id" : "OSCAP_REMEDIATE_XCCDF_ID",
-    "benchmark_id" : "OSCAP_REMEDIATE_BENCHMARK_ID",
-    "tailoring" : "OSCAP_REMEDIATE_TAILORING",
-    "tailoring_id" : "OSCAP_REMEDIATE_TAILORING_ID",
-    "arf_result" : "OSCAP_REMEDIATE_ARF_RESULT",
-    "html_report" : "OSCAP_REMEDIATE_HTML_REPORT",
-    "verbose_log" : "OSCAP_REMEDIATE_VERBOSE_LOG",
-    "verbose_level" : "OSCAP_REMEDIATE_VERBOSE_LEVEL",
-}
+XML_RESULTS = "eval_remediate_results.xml"
+HTML_REPORT = "eval_remediate_report.html"
 
 def main(tree, options):
-    filename = options["filename"]
+    # required vars
     config = options["config"]
-
-    with open(f"{tree}/{filename}", "w", encoding="utf-8") as f:
-        for ours, theirs in OSCAP_OPTIONS.items():
-            value = config.get(ours)
-            if value:
-                f.write(f"{theirs}={value}\n")
-
-    os.symlink(f"{filename}", f"{tree}/system-update")
+    profile = config["profile_id"]
+    datastream = config["datastream"]
+    # optional vars
+    path = options.get("path")
+    xccdf_id = config.get("xccdf_id")
+    ds_id = config.get("datastream_id")
+    tailoring = config.get("tailoring")
+    xml_results = config.get("xml_results", XML_RESULTS)
+    html_report = config.get("html_report", HTML_REPORT)
+
+    # run openscap in chroot on the image tree
+    cmd = [
+      "/usr/sbin/chroot", tree,
+      "/usr/bin/oscap", "xccdf", "eval",
+      "--remediate", "--profile", profile
+    ]
+
+    if path is not None:
+        os.makedirs(f"{tree}/{path.lstrip('/')}", exist_ok=True)
+        # run with command in chroot so full path ok
+        cmd.extend(["--results", f"{path}/{xml_results}"])
```

Comment:
I would have expected the default to be to not write any reports to the image itself. How come we are doing this? Could the default be to not do so and we could enable it (if desired) from composer?

Another thought: might it make sense to return the results as osbuild metadata? Not sure if we want to map it to JSON or just return an opaque blob with the xml, depends on the size/format I guess. I could imagine it may be (eventually) useful to display this in the UI as a property of a built image. Probably want that to be in a follow-up (if it makes sense at all), so not suggesting you introduce that in this PR.

Reply:
So yeah this is optional and by default we won't write any reports to the image. If we specify the path it will be. They are writing the results to the image by default in the anaconda install.

But yeah returning the results so we can use it as metadata sounds like a good shout.

Reply:
D'oh, I can't read. Ignore me :)